### PR TITLE
Rewrite: CSS namespace module and add CSS & HTML to glossary

### PR DESCRIPTION
**Pull request by estelle**
Part of the CSS modules project

* add explanation, related concepts, guides, and see also to CSS module page
* add html/xml/css/svg to namespace glossary

### CODE REVIEW

**Comment by dipikabh** [comment]
Sorry, this missed my radar. Taking a look at this now.

File: files/en-us/glossary/namespace/index.md [hunk 1]
```diff
@@ -8,8 +8,14 @@ page-type: glossary-definition
...
 
-In an operating system a directory is a namespace. Each file or subdirectory has a unique name, but one file may use the same name multiple times.
+In an operating system, a directory is a namespace. Each file or subdirectory has a unique name, but one file may use the same name multiple times.
+
+In HTML, CSS, and XML-based languages, a namespace is the explicitly declared or implied dialect to which an element (or attribute) belongs.
 
 ## See also
 
+- [Namespaces crash course](/en-US/docs/Web/SVG/Namespaces_Crash_Course)
+- [CSS namespaces module](/en-US/docs/Web/CSS/CSS_namespaces)
```
Comment (dipikabh) [inline files/en-us/glossary/namespace/index.md:18]:
```suggestion
- [CSS namespaces](/en-US/docs/Web/CSS/CSS_namespaces) module
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 2]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
 
 {{CSSRef}}
 
-The **CSS namespaces** allows authors to specify [XML namespaces](/en-US/docs/Web/SVG/Namespaces_Crash_Course) in CSS.
+The **CSS namespaces** module defines the syntax for using {{glossary("namespace", "namespaces")}} in CSS.
+
+CSS isn't just for HTML. A single stylesheet may be used to style SVG, MathML, XML, HTML, each of which has a different namespace, or a document containing multiple namespaces. Element tag names are not all unique to a single language. For example, The `<a>` element isn't limited to HTML. You may want to use different CSS style for your SVG differently than for your HTML links, or ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. [The `@namespace` rule](/en-US/docs/Web/CSS/@namespace), defined in this module, enable distinguishing between same-named elements in different namespaces.
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:12]: this needs rephrasing "You may want to use different CSS style for your SVG differently than for your HTML links"

```suggestion
CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to use different CSS style for your SVG differently than for your HTML links or ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces.
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 3]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
...
+
+The `@namespace` rule is used for declaring a default namespace and for binding namespaces to namespace prefixes. The namespaces module also defines the syntax for using these prefixes to represent namespace-qualified names. That's it. What a name means, or if the name is even valid, depends on the context and host language.
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:14]:
```suggestion
The `@namespace` rule is used for declaring a default namespace and for binding namespaces to namespace prefixes. The namespaces module also defines the syntax for using these prefixes to represent namespace-qualified names. That's it. What a name means or if the name is even valid depends on the context and host language.
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 4]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
...
+
+- [Namespaces crash course](/en-US/docs/Web/SVG/Namespaces_Crash_Course)
+
+  - : Deep dive into what a namespace is and how they are used in XML and XML-based markup languages.
+
+## Related concepts
+
+- CSS [namespace separator (`|`)](/en-US/docs/Web/CSS/Namespace_separator) combinator
+- [CSS type selector](/en-US/docs/Web/CSS/Type_selectors)
+- [CSS universal selector](/en-US/docs/Web/CSS/Universal_selectors)
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:32]:
```suggestion
- CSS [type selectors](/en-US/docs/Web/CSS/Type_selectors)
- CSS [universal selector](/en-US/docs/Web/CSS/Universal_selectors)
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 5]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
...
+- [`Element.getElementsByTagNameNS()`](/en-US/docs/Web/API/Element/getElementsByTagNameNS) method
+- [`Element.hasAttributeNS()`](/en-US/docs/Web/API/Element/hasAttributeNS) method
+- [`Element.namespaceURI`](/en-US/docs/Web/API/Element/namespaceURI) property
+- [`Element.removeAttributeNS()`](/en-US/docs/Web/API/Element/removeAttributeNS) method
+- [`Element.setAttributeNS()`](/en-US/docs/Web/API/Element/setAttributeNS) method
+- [`Element.setAttributeNodeNS()`](/en-US/docs/Web/API/Element/setAttributeNodeNS) method
+- [`NamedNodeMap.getNamedItemNS()`](/en-US/docs/Web/API/NamedNodeMap/getNamedItemNS) method
+- [`NamedNodeMap.removeNamedItemNS()`](/en-US/docs/Web/API/NamedNodeMap/removeNamedItemNS) method
+- [`NamedNodeMap.setNamedItemNS()`](/en-US/docs/Web/API/NamedNodeMap/setNamedItemNS) method
+- {{glossary("namespace")}} glossary term
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:50]:
```suggestion
- {{glossary("Namespace")}} glossary term
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 6]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
...
 
 {{Compat}}
+
+## See also
+
+- [CSS Namespace example in SVG](/en-US/docs/Web/SVG/Element/a#example)
+- [CSS `url()` functions](/en-US/docs/Web/CSS/url)
+- [CSS at-rules](/en-US/docs/Web/CSS/At-rule)
+- [CSS at-rule functions](/en-US/docs/Web/CSS/At-rule-functions)
+- [CSS selector module](/en-US/docs/Web/CSS/CSS_selectors)
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:66]: I realize that this is not defined in the "See also" guidelines, but in this case should we add a lead-in for this list, something like "These are a few examples of different namespaces:"?

```suggestion
- [`<a>`](/en-US/docs/Web/SVG/Element/a#example) SVG element
- [CSS `url()` function](/en-US/docs/Web/CSS/url)
- [CSS at-rules](/en-US/docs/Web/CSS/At-rule)
- [CSS at-rule functions](/en-US/docs/Web/CSS/At-rule-functions)
- [CSS selectors](/en-US/docs/Web/CSS/CSS_selectors)
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks for making these updates, @estelle!

File: files/en-us/web/css/css_namespaces/index.md [hunk 7]
```diff
@@ -7,18 +7,60 @@ browser-compat: css.at-rules.namespace
 
 {{CSSRef}}
 
-The **CSS namespaces** allows authors to specify [XML namespaces](/en-US/docs/Web/SVG/Namespaces_Crash_Course) in CSS.
+The **CSS namespaces** module defines the syntax for using {{glossary("namespace", "namespaces")}} in CSS.
+
+CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to use different CSS style for your SVG differently than for your HTML links or ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces.
```
Comment (estelle) [inline files/en-us/web/css/css_namespaces/index.md:12]:
```suggestion
CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces. 

The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to style the `<a>`s  within your SVGs differently than links in your HTML. You also likely want to ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. Namespacing can help.
```

File: files/en-us/glossary/namespace/index.md [hunk 8]
```diff
@@ -8,8 +8,14 @@ page-type: glossary-definition
 
 Namespace is a context for identifiers, a logical grouping of names used in a program. Within the same context and same scope, an identifier must uniquely identify an entity.
 
-In an operating system a directory is a namespace. Each file or subdirectory has a unique name, but one file may use the same name multiple times.
+In an operating system, a directory is a namespace. Each file or subdirectory within a directory has a unique name; the same name can be used multiple times across subdirectories."
```
Comment (dipikabh) [inline files/en-us/glossary/namespace/index.md:11]:
```suggestion
In an operating system, a directory is a namespace. Each file or subdirectory within a directory has a unique name; the same name can be used multiple times across subdirectories.
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 9]
```diff
@@ -7,18 +7,62 @@ browser-compat: css.at-rules.namespace
...
+The **CSS namespaces** module defines the syntax for using {{glossary("namespace", "namespaces")}} in CSS.
+
+CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces. 
+
+The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to style the `<a>`s  within your SVGs differently than links in your HTML. You also likely want to ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. Namespacing can help.
+
+The `@namespace` rule is used for declaring a default namespace and for binding namespaces to namespace prefixes. The namespaces module also defines the syntax for using these prefixes to represent namespace-qualified names. That's it. What a name means or if the name is even valid depends on the context and host language.
```
Comment (estelle) [inline files/en-us/web/css/css_namespaces/index.md:16]:
```suggestion
CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces.

The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to style the `<a>`s within your SVGs differently than links in your HTML. You also likely want to ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. Namespacing can help.

The `@namespace` rule is used for declaring a default namespace and for binding namespaces to namespace prefixes. The namespaces module also defines the syntax for using these prefixes to represent namespace-qualified names. That's it. What a name means or if the name is even valid depends on the context and host language.
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 10]
```diff
@@ -7,18 +7,62 @@ browser-compat: css.at-rules.namespace
 
 {{CSSRef}}
 
-The **CSS namespaces** allows authors to specify [XML namespaces](/en-US/docs/Web/SVG/Namespaces_Crash_Course) in CSS.
+The **CSS namespaces** module defines the syntax for using {{glossary("namespace", "namespaces")}} in CSS.
+
+CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces.
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md]:
```suggestion
CSS isn't just for styling HTML. A stylesheet may be used to style SVG, MathML, XML, or HTML, each of which has a different namespace or a document containing multiple namespaces.
```

File: files/en-us/web/css/css_namespaces/index.md [hunk 11]
```diff
@@ -7,18 +7,62 @@ browser-compat: css.at-rules.namespace
...
+
+The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to style the `<a>`s within your SVGs differently than links in your HTML. You also likely want to ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. Namespacing can help.
```
Comment (dipikabh) [inline files/en-us/web/css/css_namespaces/index.md:14]:
```suggestion
The [`@namespace`](/en-US/docs/Web/CSS/@namespace) at-rule defined in this module enables distinguishing between same-named elements in different namespaces. Element tag names are not unique to a single language. For example, the `<a>` element isn't limited to HTML. You may want to style the `<a>`s within your SVGs differently from the links in your HTML. You also will likely want to ensure that {{domxref("Document.querySelectorAll", "querySelectorAll(\"a\")")}} selects the right kind of element. Namespacing can help.
```

**Review by dipikabh — CHANGES_REQUESTED** [review]

**Review by dipikabh — APPROVED** [review]
Thank you 👍

**Pull request merged**